routing_common/transit_types.cpp

```diff
@@ -165,6 +165,13 @@ bool Gate::IsValid() const
 }
 
 // ShapeId ----------------------------------------------------------------------------------------
+bool ShapeId::operator<(ShapeId const & rhs) const
+{
+  if (m_stop1Id != rhs.m_stop1Id)
+    return m_stop1Id < rhs.m_stop1Id;
```

Comment:
Let's also write a test for this.
Also, are there tests for Unique() and Valid()?

Reply:
I added test on IsValid() for transit data structure at transit_types.
Besides that I fixed some issues about IsValid() and add some more strict checks for IsValid() according to agreement with @Zverik PTAL.

As for Unique these tests will be write later, with a big tests on GraphData.

Reply:
> Let's also write a test for this.

Done